app/services/commitments/minimum/in_arrears/calculate_true_up_fee_service.rb

```diff
@@ -115,6 +115,53 @@ def charge_in_advance_recurring_fees
 
           scope
         end
+
+        def fixed_charge_fees
+          dates_service = Commitments::DatesService.new_instance(
+            commitment: minimum_commitment,
+            invoice_subscription:
+          ).call
+
+          Fee
+            .fixed_charge
+            .joins(:fixed_charge)
+            .where(
+              subscription_id: subscription.id,
+              fixed_charge: {pay_in_advance: false}
+            )
+            .where(
+              "(fees.properties->>'fixed_charges_from_datetime') >= ?",
+              dates_service.previous_beginning_of_period
+            )
+            .where(
+              "(fees.properties->>'fixed_charges_to_datetime') <= ?",
+              dates_service.end_of_period&.iso8601(3)
+            )
+        end
```

Comment:
Would it make sense to extract a method that contains most of the logic from this method and `charge_fees` and just have:

```
def fixed_charge_fees()
   fees(fixed: true)
end

def charge_fees()
   fees(fixed: false)
end

```